### PR TITLE
Reuse MinObservation helper in commit.

### PR DESCRIPTION
**Pull request by winder**
Remove duplicate observation counting code.

### CODE REVIEW

File: shared/consensus.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+// GetConsensusMap takes a mapping from chains to a list of items,
+// return a mapping from chains to a single consensus item.
+// The consensus item for a given chain is the item with the
+// most observations that was observed at least fChain times.
+func GetConsensusMap[T any](
+	lggr logger.Logger,
+	objectName string,
+	itemsByChain map[cciptypes.ChainSelector][]T,
+	minObs map[cciptypes.ChainSelector]int,
+) map[cciptypes.ChainSelector]T {
```
Comment (makramkd) [inline shared/consensus.go:19]: A test for this would be nice

File: commit/merkleroot/outcome_test.go [hunk 2]
```diff
@@ -36,78 +34,3 @@ func Test_buildReport(t *testing.T) {
 		}
 	})
 }
-
-func Test_fChainConsensus(t *testing.T) {
```
Comment (makramkd) [inline commit/merkleroot/outcome_test.go:40]: Maybe we can copy/paste this test to the shared folder?

File: execute/plugin_functions.go [hunk 3]
```diff
@@ -227,9 +227,9 @@ func mergeMessageObservations(
 	aos []decodedAttributedObservation, fChain map[cciptypes.ChainSelector]int,
 ) (exectypes.MessageObservations, error) {
 	// Create a validator for each chain
-	validators := make(map[cciptypes.ChainSelector]validation.MinObservationFilter[cciptypes.Message])
```
Comment (makramkd) [inline execute/plugin_functions.go:230]: If this is moved to `filter`, can it be axed from the `validation` package in exec?

Comment (winder) [reply execute/plugin_functions.go:230]: Thanks, I missed that. Yes, that was the intention.

Comment (makramkd) [reply execute/plugin_functions.go:230]: Ah you're right, I don't know why I expected the files to show up as deleted, they'd be just moved.

**Review by makramkd — APPROVED** [review]

File: commit/merkleroot/outcome_test.go (hunk 2 above)
Comment (winder) [reply commit/merkleroot/outcome_test.go:40]: done.

**Pull request merged**